app/livechat/server/api/v1/room.js

```diff
@@ -179,3 +181,48 @@ API.v1.addRoute('livechat/room.forward', { authRequired: true }, {
 		API.v1.success(Meteor.runAsUser(this.userId, () => Meteor.call('livechat:transfer', this.bodyParams)));
 	},
 });
+
+API.v1.addRoute('livechat/room.visitor', { authRequired: true }, {
+	put() {
+		try {
+			check(this.bodyParams, {
+				rid: String,
+				visitorId: String,
```

Comment:
My suggestion is to request not just the new visitor but also the old one, and then check if the current room visitor and the old parameter match. 
```suggestion
				oldVisitorId: String,
				newVisitorId: String,
```